File: frontend/src/types/rental.ts
export interface RentalProfile {
  locality: string;
  postcodes: string[];
  dwelling_type: string;
  bedrooms: number | null;
  median_weekly_rent: number | null;
  lower_quartile_rent: number | null;
  upper_quartile_rent: number | null;
  lodgement_count: number;
  confidence: "low" | "medium" | "high";
  rent_growth_1y_pct: number | null;
  median_sale_price: number | null;
  gross_yield_pct: number | null;
  data_as_of: string | null;
}

export interface RentalTrendPoint {
  month: string;
  median_weekly_rent: number;
  lodgement_count: number;
  confidence: "low" | "medium" | "high";
}

export interface RentalScreenResult {
  locality: string;
  postcodes: string[];
  median_weekly_rent: number;
  median_sale_price: number;
  gross_yield_pct: number;
  rental_count: number;
  sales_count: number;
  confidence: "low" | "medium" | "high";
}

export interface RentalFilters {
  dwellingType?: string;
  bedrooms?: string;
}

File: backend/models/rental.py
"""Rental investment analytics models."""

from dataclasses import dataclass
from datetime import date


@dataclass(frozen=True)
class RentalProfile:
    locality: str
    postcodes: list[str]
    dwelling_type: str
    bedrooms: int | None
    median_weekly_rent: float | None
    lower_quartile_rent: float | None
    upper_quartile_rent: float | None
    lodgement_count: int
    confidence: str
    rent_growth_1y_pct: float | None
    median_sale_price: float | None
    gross_yield_pct: float | None
    data_as_of: date | None


@dataclass(frozen=True)
class RentalTrendPoint:
    month: date
    median_weekly_rent: float
    lodgement_count: int
    confidence: str


@dataclass(frozen=True)
class RentalScreenResult:
    locality: str
    postcodes: list[str]
    median_weekly_rent: float
    median_sale_price: float
    gross_yield_pct: float
    rental_count: int
    sales_count: int
    confidence: str
